pilot/pkg/networking/plugin/authz/rbac_v2.go

```diff
@@ -77,6 +85,15 @@ func convertRbacRulesToFilterConfigV2(service *serviceMetadata, option rbacOptio
 			if binding.RoleRef != nil {
 				role = option.authzPolicies.RoleForNameAndNamespace(binding.RoleRef.Name, namespace)
 				serviceRoleName = binding.RoleRef.Name
+			} else if binding.Role != "" {
+				if strings.HasPrefix(binding.Role, serviceRoleRootNamespaceRef) {
+					localRoleName := binding.Role[1:]
+					role = option.authzPolicies.RoleForNameAndNamespace(localRoleName, model.DefaultMeshConfig().RootNamespace)
+					serviceRoleName = localRoleName
+				} else {
+					role = option.authzPolicies.RoleForNameAndNamespace(binding.Role, namespace)
+					serviceRoleName = binding.Role
+				}
```

Comment:
Can you add some test cases for the code added here?

Reply:
I'm concerned about adding more unit tests to `rbac_v2_test.go`. It's much more verbose than unit tests from `rbac_test.go`. I tested this PR manually and it should be fine. Ideally, I want to wait for https://github.com/istio/istio/pull/12814 to be checked in, so I can add e2e tests, and wait for @yangminzhu to refactor `rbac_test.go`, so I can apply the new testing methods to `rbac_v2_test.go`. 

@yangminzhu Are you working on refactoring `rbac_test.go`? Thanks.

Reply:
You can add the tests right now, no need to wait for the refactor.

Reply:
Added a unit test.